torch/distributed/fsdp/_utils.py

```diff
@@ -2,13 +2,47 @@
 
 import torch
 from torch.nn.modules.batchnorm import _BatchNorm
+from torch.nn.modules.normalization import LayerNorm
 from torch.utils._mode_utils import no_dispatch
 
+from torch.distributed.utils import _apply_to_tensors
+from functools import partial
+
 
 def _contains_batchnorm(module):
     return any(isinstance(mod, _BatchNorm) for mod in module.modules())
 
 
+def _contains_module(root, module_to_check):
+    return any(isinstance(mod, module_to_check) for mod in root.modules())
+
+def _override_module_mixed_precision(root, module_to_override, wrap_override_dict={"mixed_precision": None}):
+    for mod in root.modules():
+        if isinstance(mod, module_to_override):
+            mod._wrap_overrides = wrap_override_dict  # type: ignore[assignment]
+
+        if isinstance(mod, LayerNorm): # TODO: generalize this logic if additional types need to be supported
+
+            old_dtype = None
```

Comment:
Wonder what we can do here. The input -> output mapping for a module is not necessarily 1:1. We're basically saving an `old_dtype` so that we cast the output back to whatever the input precision was before forcing to fp32. But agreed there can be multiple input precisions.

Maybe we don't cast back at all (so we don't need a post hook or old dtype), and rely on users doing `cast_forward_inputs` configuring correctly to ensure dtypes into subsequent mixed precision modules are correct?